sphinx_gallery/gen_rst.py

```diff
@@ -208,6 +208,18 @@ def _sanitize_rst(string):
     string = re.sub(p + r'`([^`]+)`' + e, r'\1\2\3', string)
     # `whatever thing` --> whatever thing
     string = re.sub(p + r'([^`]+)' + e, r'\1\2\3', string)
+
+    # **string** --> string
+    string = re.sub(r'\*\*([^\*]*)\*\*', r'\1', string)
+    # *string* --> string
+    string = re.sub(r'\*([^\*]*)\*', r'\1', string)
+    # `link text <url>`_ --> link text
+    string = re.sub(r'`(.*) <.*>`\_', r'\1', string)
+    # :term:`the term` --> the term
+    string = re.sub(r':term:`(.*)`', r'\1', string)
+    # :ref:`the ref` --> the ref
+    string = re.sub(r':ref:`(.*)`', r'\1', string)
```

Comment:
There are many possible roles here, so perhaps we want something more like
```suggestion
    # :role:`the thing` --> the thing
    string = re.sub(r':[a-z]+:`([^`]+)`', r'\1', string)
```
Note the generalization of the `:role:` and also the exclusion of any backticks within the captured group -- otherwise something like this would probably be an issue because regex is greedy:
```
:meth:`a.B` and :meth:`a.C`
```
I think would collapse to something bad like:
```
a.b` and :meth:`a.C
```
As another issue, people can also do
```
:meth:`My method <a.B>`
```
but we could deal with that later/in another PR unless you're keen to work on more regex :)

Reply:
Actually I think this works:
```
:[a-z]+:`([^`<>]+)( <[^`<>]+>)?`
```
On https://regex101.com/ it replaces
```
:meth:`MyMethod <a.B>` and :meth:`a.C`
```
with 
```
MyMethod and a.C
```

Reply:
I don't think it's a good idea to implement a (subset of a) docutils parser by hand. Why not use existing code? Doesn't Sphinx have some functionality like this?

As a concrete example what can go wrong when doing it manually: did you know that the role name can also be used as a suffix?
https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#interpreted-text

Reply:
Agreed it would be good to use something built in if we can.

In the meantime I'd rather not let the perfect be the enemy of the good here. In years of looking through intersphinx refs I've seen hundreds or thousands of prefixes and no postfixes. It's not saying they don't exist, but again, perfect being the enemy of the good... I'd assume YAGNI on trying to catch every variant here if we do indeed need to start with our own parsing.

And I'm okay with merging this imperfect/manual version as-is, with a follow-up PR to improve it. This already improves on what we have I think.